### PR TITLE
Fix grammar typo in GDScript error message

### LINKED ISSUE #83453: Grammar typo in GDScript error message
### Godot version

Godot v4.2.beta1 - Pop!_OS 22.04 LTS - X11 - Vulkan (Forward+) - integrated Intel(R) Graphics (ADL GT2) () - 12th Gen Intel(R) Core(TM) i5-1235U (12 Threads)

### System information

Godot v4.2.beta1 - Pop!_OS 22.04 LTS - X11 - Vulkan (Forward+) - integrated Intel(R) Graphics (ADL GT2) () - 12th Gen Intel(R) Core(TM) i5-1235U (12 Threads)

### Issue description

Easy bugfix for grabs!

image

> a "int"

The "a" is not correct here, which is convenient as it removes the need to worry between "a" and "an".

### Steps to reproduce

Not relevant

### Minimal reproduction project

N/A

### PR DESCRIPTION
**Pull request by Lunarisnia**
Fixes #83453

### CODE REVIEW

**Comment by akien-mga** [comment]
The point of that issue was to remove "a", not change it to "an".

File: modules/gdscript/gdscript_analyzer.cpp [hunk 1]
```diff
@@ -4336,7 +4336,7 @@ void GDScriptAnalyzer::reduce_subscript(GDScriptParser::SubscriptNode *p_subscri
 					}
 				} else if (base_type.kind != GDScriptParser::DataType::BUILTIN && !index_type.is_variant()) {
 					if (index_type.builtin_type != Variant::STRING && index_type.builtin_type != Variant::STRING_NAME) {
-						push_error(vformat(R"(Only String or StringName can be used as index for type "%s", but received a "%s".)", base_type.to_string(), index_type.to_string()), p_subscript->index);
+						push_error(vformat(R"(Only String or StringName can be used as index for type "%s", but received an "%s".)", base_type.to_string(), index_type.to_string()), p_subscript->index);
```
Comment (dalexeev) [inline modules/gdscript/gdscript_analyzer.cpp:4339]:
```suggestion
						push_error(vformat(R"(Only "String" or "StringName" can be used as index for type "%s", but received "%s".)", base_type.to_string(), index_type.to_string()), p_subscript->index);
```

Comment (Lunarisnia) [reply modules/gdscript/gdscript_analyzer.cpp]: Thank you!

**Comment by dalexeev** [comment]
Please use a local git client to [squash](https://docs.godotengine.org/en/stable/contributing/workflow/pr_workflow.html#the-interactive-rebase) comments into one. The GitHub web interface does not allow this, as far as I know.

**Review by dalexeev — APPROVED** [review]

**Pull request merged**